Library/Homebrew/cask/artifact/abstract_uninstall.rb

```diff
@@ -130,21 +130,52 @@ def uninstall_quit(*bundle_ids, command: nil, **_)
             next
           end
 
-          ohai "Quitting application ID '#{bundle_id}'."
-          command.run!("/usr/bin/osascript", args: ["-e", %Q(tell application id "#{bundle_id}" to quit)], sudo: true)
+          ohai "Quitting application '#{bundle_id}'..."
 
           begin
-            Timeout.timeout(3) do
+            Timeout.timeout(10) do
               Kernel.loop do
-                break if running_processes(bundle_id, command: command).empty?
+                next unless quit(bundle_id).success?
+                if running_processes(bundle_id, command: command).empty?
+                  puts "Application '#{bundle_id}' quit successfully."
+                  break
+                end
               end
             end
           rescue Timeout::Error
+            opoo "Application '#{bundle_id}' did not quit."
             next
           end
         end
       end
 
+      def quit(bundle_id)
+        script = <<~JAVASCRIPT
+          'use strict';
+
+          ObjC.import('stdlib')
+
+          function run(argv) {
+            var app = Application(argv[0])
+
+            try {
+              app.quit()
+            } catch (err) {
+              if (app.running()) {
+                $.exit(1)
+              }
+            }
+
+            $.exit(0)
+          }
+        JAVASCRIPT
+
+        system_command "osascript", args:         ["-l", "JavaScript", "-e", script, bundle_id],
```

Comment:
```suggestion
        command.run "osascript", args:         ["-l", "JavaScript", "-e", script, bundle_id],
```

Reply:
I'm actually trying to get rid of everything other than `system_command`, i.e. unify all the `system`, `quiet_system` and `command.run` calls.

Reply:
How are we going to deal with the numerous tests that rely on `fake_system_command`?

Reply:
Refactor them as we go. The ones using the `FakeSystemCommand` class are not idiomatic RSpec anyways.

Reply:
You’re right; your solution is indeed more idiomatic. It also makes both the tests and the code under test simpler and easier to read! 👍